src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexAssemblyCompiler.cs

```diff
@@ -65,13 +65,13 @@ internal void GenerateRegexType(string pattern, RegexOptions options, string nam
             Type runnerType = regexRunnerTypeBuilder.CreateType()!;
 
             // Generate the RegexRunnerFactory-derived type.
-            TypeBuilder regexRunnerFactoryTypeBuilder = DefineType(_module, $"{name}Factory{typenumString}", false, typeof(RegexRunnerFactory));
+            TypeBuilder regexRunnerFactoryTypeBuilder = DefineType(_module, $"{name}Factory{typenumString}", isPublic: false, isSealed: true, typeof(RegexRunnerFactory));
             _ilg = DefineMethod(regexRunnerFactoryTypeBuilder, "CreateInstance", typeof(RegexRunner));
             GenerateCreateInstance(runnerType);
             Type regexRunnerFactoryType = regexRunnerFactoryTypeBuilder.CreateType()!;
 
             // Generate the Regex-derived type.
-            TypeBuilder regexTypeBuilder = DefineType(_module, name, isPublic, typeof(Regex));
+            TypeBuilder regexTypeBuilder = DefineType(_module, name, isPublic, isSealed: false, typeof(Regex));
```

Comment:
Remind me why this can't be sealed? I thought we were making 3 types here that were derived, but were not derived from (maybe worth a comment)

Reply:
Because it's almost always public (you can't use it if it's not, so passing isPublic: false is silly; the docs say as much).  We could seal it if we wanted to, but it'd be a change from netfx, where you could derive from the generated type.

Reply:
OK